pkg/consensus/payload.go

```diff
@@ -195,18 +196,34 @@ func (p *Payload) Sign(key *privateKey) error {
 }
 
 // Verify verifies payload using provided Witness.
-func (p *Payload) Verify() bool {
-	h := sha256.Sum256(p.MarshalUnsigned())
+func (p *Payload) Verify(scriptHash util.Uint160) bool {
+	verification := p.Witness.VerificationScript
+	if len(verification) == 0 {
+		bb := new(bytes.Buffer)
+		err := vm.EmitAppCall(bb, scriptHash, false)
+		if err != nil {
+			return false
+		}
+		verification = bb.Bytes()
+	} else if h := p.Witness.ScriptHash(); scriptHash != h {
+		return false
+	}
```

Comment:
This chunk of code obviously reminds me of `verifyHashAgainstScript()`, I think we can make this code more reusable.